### PR TITLE
Embed and Key Creation examples

### PR DESCRIPTION
**Pull request by postables**
I've included an example of creating an IPNS entry and embedding a public key into it, as well as generating an RSA and ED25519 keypair. I've included test files as well.

### CODE REVIEW

File: examples/examples_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return kp, nil
+}
+
+func generateEDKey() (*examples.KeyPair, error) {
+	// DO NOT USE 1024 BITS IN PRODUCTION
+	// THIS IS ONLY FOR TESTING PURPOSES
+	kp, err := examples.GenerateEDKeyPair(1024)
```
Comment (Stebalien) [inline examples/examples_test.go]: FYI, ed25519 actually just ignores the bits. It always uses 256 bit public keys.

The "use >=2048 bit keys" advice only applies to RSA. Other algorithms have different minimums.

File: examples/key.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package examples
+
+import (
+	crypto "github.com/libp2p/go-libp2p-crypto"
+)
+
+// KeyPair is a helper struct used to contain the parts of a key
+type KeyPair struct {
+	Private crypto.PrivKey
```
Comment (Stebalien) [inline examples/key.go]: FYI, PrivKey contains the public key as well (you can call `GetPublic()` on it to get the public part).

Comment (postables) [reply examples/key.go]: Good point!

File: examples/examples_test.go (hunk 1 above)
Comment (postables) [reply examples/examples_test.go]: Interesting, this is something I will do more reading about.

**Comment by Stebalien** [comment]
Thanks!

**Pull request merged**